### DISCUSSION TITLE
128 word Optiboot for ATmega88/168 (assembler rewrite)

### DISCUSSION THREAD

**Discussion by bitflipser** [Optiboot/optiboot#311] [Show and tell]
https://github.com/bitflipser/miniOptiboot

**Comment by WestfW** [comment]
Bitflipser has produced a couple of  Atmel-assembler bootloaders based on the "structure" and "philosophy" of Optiboot.  This version cuts the final size down to the "smallest possible" bootloader on atmega168 and atmega88 chips (128 words.)
There is also "ASMboot" at https://github.com/bitflipser/ASMboot which fits additional features into the 256-word minimum size of ATmega328.  (EEPROM Read/Write, real fuse/signature reading, "LPM callable from application", and a few more.)

**Comment by davidbitton** [comment]
@WestfW I just tried and put ASMboot on my 328p board. I used the following:
```
avrdude -v -v -c usbasp -p m328p -Uflash:w:ASMboot_328_16.hex:i
```

but using the Arduino IDE to push a sketch via FTDI fails with the `avrdude: stk500_getsync()` error. Ideas?
